### PR TITLE
json config file w/ keygen ids

### PR DESCRIPTION
**Pull request by DT9**

### CODE REVIEW

File: web/bfex/common/utils.py [hunk 1]
```diff
@@ -73,5 +74,12 @@ def normalize(text):
         return normalized
 
 
+class ConfigFile:
+    def __init__(self):
+        self.data = []
+        with open('../../../../config.json') as json_config_file:
```
Comment (GarryBullock) [inline web/bfex/common/utils.py]: This should be configurable via an environment variable.

File: web/bfex/components/data_pipeline/tasks/keywords.py [hunk 2]
```diff
@@ -13,6 +13,9 @@ class UpdateKeywordsFromScrape(Task):
     """
     def __init__(self):
         self.task_name = "Update Keywords From Scrape"
+        self.approaches = [RakeApproach(), GenericApproach()]
+        self.ids = ConfigFile().data['keygen_ids']
+        self.filtered_algs = [alg for alg in self.approaches if alg.get_id() in self.ids]
```
Comment (GarryBullock) [inline web/bfex/components/data_pipeline/tasks/keywords.py]: You arent actually using the KeywordGenerator here. We should be using that, register each of the configured approaches with the KeyGenerator so we can reuse it.

Lane is using the KeyGenerator in #37 maybe take a look at that and work with him to get this functionality working.

File: web/bfex/common/singleton_decorator.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# https://stackoverflow.com/questions/31875/is-there-a-simple-elegant-way-to-define-singletons
```
[the remark this answers is not in the document]
Comment (DT9) [reply web/bfex/common/singleton_decorator.py]: https://stackoverflow.blog/2009/06/25/attribution-required/
Thanks, caught my error

File: web/bfex/components/key_generation/__init__.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+__all__ = []
+
+import pkgutil
+import inspect
+
+for loader, name, is_pkg in pkgutil.walk_packages(__path__):
+    module = loader.find_module(name).load_module(name)
+
```
Comment (GarryBullock) [inline web/bfex/components/key_generation/__init__.py]: Where did this code come from? Did you write it yourself? I like it a lot.

Comment (DT9) [reply web/bfex/components/key_generation/__init__.py]: Yea... I need to get used to that more often

**Pull request merged**